tests/utils.py

```diff
@@ -13,6 +14,8 @@ async def run_server(config: Config, sockets=None):
     await asyncio.sleep(0.1)
     try:
         yield server
+    except BaseException:  # pragma: no cover
+        traceback.print_exc()
```

Comment:
This is actually to have more visibility.

Let me explain a bit... Well... The `run_server` is a context manager that prevents the test to fail. We need to refactor this logic, or only have assertions outside the `run_server`. In any case, the idea here is just to see the potential exceptions when running `pytest` with `-s`.